plugins/services/src/js/reducers/serviceForm/JSONReducers/Volumes.js

```diff
@@ -179,8 +179,16 @@ module.exports = {
       return [];
     }
 
-    return state.container.volumes.reduce(function(memo, item, index) {
-      /**
+    return state.container.volumes
+      .filter(function(item) {
+        return (
```

Comment:
It is not clear to me that this function excludes unknown volumes. Especially because I see 4 types of volumes, and 3 conditions. Persistent Volume is persistent. External Volume is external, so I am left to assume that both DC/OS Storage and Host Volume have a hostPath. 

Maybe a comment that explains why this is enough to define "known volume" would be helpful.

Reply:
So DCOS storage volumes are also persistent volumes. :-) 
Host volumes do not have specific field except hostPath

For a better overview you can look at https://github.com/mesosphere/marathon/blob/master/docs/docs/rest-api/public/api/v2/types/volumes.raml